lib/iris/_concatenate.py

```diff
@@ -1090,6 +1103,10 @@ def concatenate(self):
 
             # Concatenate the new data payload.
             data = self._build_data()
+            if data is None:
+                shape = [len(coord_dim[0].points) for coord_dim in dim_coords_and_dims]
```

Comment:
Get the `shape` of the coord, rather than accessing `point` data? This is maybe slightly more idiomatic:

```suggestion python
                shape = [coord.shape[0] for coord, _dim in dim_coords_and_dims]
```

I'll leave it at to your discretion!